### ISSUE TITLE
Provide list of error messages and "codes"

### ISSUE THREAD

**Issue by EHJ-52n** [52North/SOS#252]
We should provide a list of errors and which error message are returned by the SOS.

**Comment by autermann** [comment]
Well, the `exceptionCode`s are already listed in the OWS and SOS spec... You don't mean the exact text messages by *error messages*, right? That would be quite a bit of work:

```sh
$ grep "throw " **/src/main/**/*.java | sort | uniq | wc -l
912
```

And these do not include `IOException`s, `HibernateException`s or `XmlException`s (think of all the possible validation errors), that are just wrapped in a `NoApplicableCodeException`, you can't possibly list these...

I don't think this effort is worth it and I don't know any web service that lists all of it's error codes. They may list major categories (which would map to our `exceptionCode`s), but never all possible error messages.

**Comment by EHJ-52n** [comment]
Another argument pro such a list: For developers not 100% familiar with the SOS specification could implement clients more easily. _"With such a list, I could know potential sources of error before making the error."_

**Comment by shane-axiom** [comment]
I agree that the OGC specifications already dictate request formats and error codes. Clients should be implementing against the OGC specs, and if there's a mismatch between the spec and the 52n SOS the SOS should be fixed!
